lang/php/lib/avro/data_file.php

```diff
@@ -163,12 +162,13 @@ public static function is_valid_codec($codec)
   /**
    * @param AvroIO $io
    * @param AvroSchema $schema
+   * @param string $codec
    * @returns AvroDataIOWriter
    */
-  protected function open_writer($io, $schema)
+  protected function open_writer($io, $schema, $codec)
```

Comment:
Shouldn't we set this one to NULL by default as well: `$codec=self::NULL_CODEC`, to maintain backward compatibility.

Reply:
Thank you for the review @fokko! I didn't set the default value to `open_writer`, since I assumed that users usually use `open_file` and don't call `open_writer` directly. But `open_writer` is not a private function, so it should have the default value to maintain backward compatibility, as you said. Will update the PR.